substrate/frame/democracy/src/lib.rs

```diff
@@ -1190,6 +1197,74 @@ pub trait EncodeInto: Encode {
 }
 impl<T: Encode> EncodeInto for T {}
 
+#[cfg(any(feature = "try-runtime", test))]
+impl<T: Config> Pallet<T> {
+	/// Ensure the correctness of the state of this pallet.
+	///
+	/// This should be valid before or after each state transition of this pallet.
+	pub fn do_try_state() -> Result<(), sp_runtime::TryRuntimeError> {
+		// PublicPropCount should be greater than PublicProps count
+		ensure!(
+			PublicProps::<T>::get().len() <= <PublicPropCount<T>>::get() as usize,
+			"`PublicPropCount` should be greater than of `PublicProps` in storage"
+		);
+		// DepositOf should be equal to PublicProps in storage
+		ensure!(
+			DepositOf::<T>::iter().count() == PublicProps::<T>::get().len() as usize,
+			"`DepositOf` should be greater than of `PublicProps` in storage"
+		);
+		// Total number of depositors should be greater or equal to `PublicProps`
+		ensure!(
+			Self::count_total_depositors() >= PublicProps::<T>::get().len() as u32,
+			"Total depositors should be greater than that of `PublicProps` in storage"
+		);
```

Comment:
I don't get this one. If we want to enforce some property on the number of depositor for a each `DespositOf` items then we should rather ensure they are all more than zero, but I think we don't need to enforce a property here.

Reply:
so, should i go ahead to remove this invariant check?